### PR TITLE
Connectionless load stac

### PR DESCRIPTION
**Pull request by EmileSonneveld**
This allows to use `openeo.rest.datacube.DataCube.load_stac(...)`
This way, we can use the python client to generate process graphs without an openeo backend.

### CODE REVIEW

File: openeo/rest/datacube.py [hunk 1]
```diff
@@ -259,6 +260,36 @@ def load_disk_collection(cls, connection: Connection, file_format: str, glob_pat
...
 
+    @classmethod
+    def load_stac(
+        cls,
+        url: str,
+        spatial_extent: Union[Dict[str, float], Parameter, None] = None,
+        temporal_extent: Union[Sequence[InputDate], Parameter, str, None] = None,
+        bands: Optional[List[str]] = None,
+        properties: Optional[Dict[str, Union[str, PGNode, Callable]]] = None,
+        connection: Connection = None,
```
Comment (soxofaan) [inline openeo/rest/datacube.py:271]:
```suggestion
        connection: Optional[Connection] = None,
```

File: tests/rest/test_connection.py [hunk 2]
```diff
@@ -2443,6 +2443,16 @@ def test_basic(self, con120):
...
+    def test_basic_connectionless(self, con120):
+        cube = openeo.DataCube.load_stac("https://provider.test/dataset")
+        assert cube.flat_graph() == {
+            "loadstac1": {
+                "process_id": "load_stac",
+                "arguments": {"url": "https://provider.test/dataset"},
+                "result": True,
+            }
+        }
+
```
Comment (soxofaan) [inline tests/rest/test_connection.py:2455]: This does not really belong in test_connection.py as this test is about _not_ using a connection 

I would put it in
https://github.com/Open-EO/openeo-python-client/blob/master/tests/rest/datacube/test_datacube.py

File: openeo/rest/datacube.py [hunk 3]
```diff
@@ -259,6 +260,36 @@ def load_disk_collection(cls, connection: Connection, file_format: str, glob_pat
...
+    ) -> DataCube:
+        arguments = {"url": url}
```
Comment (soxofaan) [inline openeo/rest/datacube.py:273]: Can you add documentation too (can be copy of Connection.load_stac)

including 

```
.. versionadded:: 0.33.0
```

**Review by soxofaan — COMMENTED** [review]
nice

**Comment by soxofaan** [comment]
can you add a changelog item too?

File: openeo/rest/datacube.py (hunk 1 above)
Comment (EmileSonneveld) [reply openeo/rest/datacube.py]: True, updated in DataCube.load_collection too

File: openeo/rest/datacube.py (hunk 3 above)
Comment (EmileSonneveld) [reply openeo/rest/datacube.py:370]: Added. Also added 'connection' to the docstring

File: tests/rest/test_connection.py (hunk 2 above)
Comment (EmileSonneveld) [reply tests/rest/test_connection.py]: Moved. And made it more explicit that the processgraph should be the same as the version with connection

File: tests/rest/datacube/test_datacube.py [hunk 4]
```diff
@@ -102,6 +102,19 @@ def test_filter_temporal_basic_extent(s2cube):
...
+        "loadstac1": {
+            "process_id": "load_stac",
+            "arguments": {"url": "https://provider.test/dataset"},
+            "result": True,
+        }
+    }
+    cube = DataCube.load_stac("https://provider.test/dataset")
+    assert cube.flat_graph() == expected_graph
+    cube2 = connection.load_stac("https://provider.test/dataset")
+    assert cube2.flat_graph() == expected_graph
```
Comment (soxofaan) [inline tests/rest/datacube/test_datacube.py:116]: Can you move this  test to this class https://github.com/Open-EO/openeo-python-client/blob/3871077d53a4f2f2b7addc327c6a15fef4a45ab2/tests/rest/datacube/test_datacube.py#L85-L136
which I just added with 3871077d53a4f2f2b7addc327c6a15fef4a45ab2

File: CHANGELOG.md [hunk 5]
```diff
@@ -9,6 +9,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 
+- Added `DataCube.load_stac()` to create a datacube without needing an openeo connection.
```
Comment (soxofaan) [inline CHANGELOG.md:12]:
```suggestion
- Added `DataCube.load_stac()` to also support creating a `load_stac` based cube without a connection ([#638](https://github.com/Open-EO/openeo-python-client/issues/638))
```

File: openeo/rest/datacube.py [hunk 6]
```diff
@@ -259,6 +260,133 @@ def load_disk_collection(cls, connection: Connection, file_format: str, glob_pat
...
+        url: str,
+        spatial_extent: Union[Dict[str, float], Parameter, None] = None,
+        temporal_extent: Union[Sequence[InputDate], Parameter, str, None] = None,
+        bands: Optional[List[str]] = None,
+        properties: Optional[Dict[str, Union[str, PGNode, Callable]]] = None,
+        connection: Optional[Connection] = None,
+    ) -> DataCube:
+        """
+        Loads data from a static STAC catalog or a STAC API Collection and returns the data as a processable :py:class:`DataCube`.
+        A batch job result can be loaded by providing a reference to it.
```
Comment (soxofaan) [inline openeo/rest/datacube.py:275]:
```suggestion
```

that's not valid anymore. Can you also remove it from connection.load_stac?

Comment (soxofaan) [reply openeo/rest/datacube.py:275]: nevermind, this comes from official spec, leave it as-is

File: tests/rest/datacube/test_datacube.py (hunk 4 above)
Comment (EmileSonneveld) [reply tests/rest/datacube/test_datacube.py]: Moved!

**Comment by EmileSonneveld** [comment]
@soxofaan , ok to merge?

**Pull request merged**